docs/resources/device_action.md

```diff
@@ -0,0 +1,142 @@
+---
```

Comment:
The documentation provided seems to follow a good structure and format, but there are some minor issues that need to be addressed to make it meet the standards for Hasicorp terraform official documentation. Here's a detailed report summarizing the documentation quality issues, with clear examples, and suggestions for improvement:

1. Copyright Notice: The copyright notice is added at the beginning of the document, but it is not formatted correctly. The correct format is:

Copyright (c) YEAR COMPANY NAME. All Rights Reserved.

Replace COMPANY NAME with the actual company name, and YEAR with the release year.

Example:
Copyright (c) 2024 Dell Inc. All Rights Reserved.

2. Licensing Information: The licensing information is also added at the beginning of the document, but it is not formatted correctly. The correct format is:

Licensed under the Mozilla Public License Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://mozilla.org/MPL/2.0/. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.

Example:
Licensed under the Mozilla Public License Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://mozilla.org/MPL/2.0/. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.

3. Headings: The headings in the documentation are not formatted correctly. The correct format is:

# Heading (level 1)
## Heading (level 2)
### Heading (level 3)

Example:
# Example Usage
## Creating a Job
### Creating a Job with Cron Expression

4. Format: The code blocks in the documentation are not formatted correctly. The correct format is:

```
Code block
```

Example:
```
# Use the device datasource to get the ids of the required devices
# Their ids shall be used to run actions on them
data "ome_device" "devs" {
  filters = {
    device_service_tags = ["CZMC1T2", "4111H63"]
  }
}
```

5. Examples: The examples provided in the documentation are not formatted correctly. The correct format is:

### Example:

```
Resource Block
```

Example:
### Example:
```
resource "ome_device_action" "code_1" {
  device_ids      = data.ome_device.devs.devices[*].id
  action          = "inventory_refresh"
  job_name        = "inventory-refresh-job"
  job_description = "Job to refresh inventory of CZMC1T2 and 4111H63 devices"
  timeout         = 8
  lifecycle {
    ignore_changes = [
      timeout,
    ]
  }
}
```

6. Schema: The schema provided in the documentation is not formatted correctly. The correct format is:

## Schema

### Required

* `device_ids` (List of Number) List of id of devices on whom the action would be carried out.
* `job_name` (String) Name of the job to be created on the OME appliance that will run the action.

### Optional

* `action` (String) Action to be performed on the devices. Accepted values are `inventory_refresh`. Default value is `inventory_refresh`.
* `cron` (String) Cron expression to schedule an action in the future. If not specified, the action runs immediately on apply. Conflicts with `timeout`.
* `job_description` (String) Description of the job to be created on the OME appliance that will run the action.
* `timeout` (Number) Timeout, in minutes, for monitoring an immediately running action. Conflicts with `cron`. Default value is `10`.

### Read-Only

* `current_status` (String) Current status of the job.
* `end_time` (String) End time of the job.
* `id` (Number) ID of the job created on OME appliance for carrying out the action.
* `last_run_status` (String) Last run status of the job.
* `last_run_time` (String) Last run time of the job.
* `next_run_time` (String) Next run time of the job.
* `start_time` (String) Start time of the job.

I hope this helps! Let me know if you have any further questions.